Eplant/tutorial.md

```diff
@@ -0,0 +1,227 @@
+  - [getInitialState:](#getinitialstate)
+  - [getInitialData:](#getinitialdata)
+  - [reducer:](#reducer)
+  - [actions:](#actions)
+  - [component:](#component)
+  - [icon:](#icon)
+  - [name:](#name)
+  - [id:](#id)
+  - [description:](#description)
+  - [thumbnail:](#thumbnail)
+  - [citation:](#citation)
+  - [header:](#header)
+- [state](#state)
+  - [Atoms:](#atoms)
+  - [Hooks:](#hooks)
+  - [Simplicity:](#simplicity)
+  - [Reactivity:](#reactivity)
+- [When you create a new `view`](#when-you-create-a-new-view)
+  - [Define GetStartedView Object:](#define-getstartedview-object)
+  - [Define GetStartedView Component:](#define-getstartedview-component)
+  - [Define Tile Component:](#define-tile-component)
+  - [Summary](#summary)
+- [Eplant2](#eplant2)
+
+
+
+## Main Component
+In the root(`Eplant/main.tsx`), you can see the `Eplant` component, whchi is our main component of our application.  
+
+`main.tsx`
+```
+<React.StrictMode>
+  <Provider>
+    <BrowserRouter>
+      <Config.Provider value={defaultConfig}>
+        <Eplant />
+      </Config.Provider>
+    </BrowserRouter>
+  </Provider>
+</React.StrictMode>
+```
+
+We only have one single Route at the moment but we will have one Route for one component.  
+
+
+`Eplant.tsx`
+```
+<ThemeProvider theme={darkMode ? dark : light}>
+  <CssBaseline />
+  <Routes>
+    <Route path={rootPath}>
+      <Route index element={<MainEplant />} />
+    </Route>
+  </Routes>
+</ThemeProvider>
+```
+
+Eplant app consists of `Sidebar` and `EplantLayout` as shown in `Eplant.tsx` (`Eplant/Eplant.tsx`)
+```
+export function MainEplant() {
+  return (
+    <>
+      <Sidebar />
+      <EplantLayout />
+    </>
+  )
+}
+```
+
+
+## Eplant Layout
+Eplant Layout (`Eplant/EplantLayout.tsx`) hosts the actual views and ViewContainer (`Eplant/UI/Layout/ViewContainer/index.tsx`) is main component that does this. 
+
+`view` will be passed as a prop to the ViewContainer which will be rendered in 
+```
+<view.component
+    state={state}
+    geneticElement={gene}
+    activeData={activeData}
+    dispatch={dispatch}
+/>
+```
+
+
+### useViewData
+The `useViewDatai` function is a custom React hook designed to manage and load view data in the application. It leverages TypeScript generics to handle different types of view data, state, and actions.
+
+See `Eplant/View/viewData.ts` for more details.
+
+
+
+## View Interface
+Please see the `View` interface in `View \index.ts`
+This is the interface we use for most of the `views`
+
+The View interface is a TypeScript interface that defines the structure and functionality expected from a view component in a React application.
+
+
+### getInitialState:
+
+An optional method that returns the initial state of the view. If not defined, the state defaults to null.
+This method initializes the state when the view is first rendered.
+
+
+### getInitialData:
+
+A required method that fetches the data needed for the view based on a given gene and provides loading feedback.
+
```

Comment:
Should there be mention on error handling in this document? Such as what happens when getInitialData works improperly?